File: composeApp/src/commonMain/kotlin/org/hhp227/concafe/presentation/notification/NotificationUiState.kt
package org.hhp227.concafe.presentation.notification

import org.hhp227.concafe.domain.model.NotificationSection

data class NotificationUiState(
    val isLoading: Boolean = true,
    val errorMessage: String? = null,
    val isLoggedIn: Boolean = false,
    val unreadCount: Int = 0,
    val sections: List<NotificationSection> = emptyList()
) {
    companion object {
        fun empty(): NotificationUiState = NotificationUiState()
    }
}

File: iosApp/iosApp/Presentation/Notification/NotificationUiState.swift
//
//  NotificationUiState.swift
//  ConCafe
//
//  Created by 홍희표 on 2026/03/06.
//

import Foundation
import Shared

struct NotificationUiState {
    var isLoading: Bool = true
    var errorMessage: String?
    var isLoggedIn: Bool = false
    var unreadCount: Int32 = 0
    var sections: [NotificationSection] = []

    static let empty = NotificationUiState()
}
